pkg/buildsettings/buildSettings.go

```diff
@@ -0,0 +1,101 @@
+	CreateBOM                   bool     `json:"createBOM,omitempty"`
+	LogSuccessfulMavenTransfers bool     `json:"logSuccessfulMavenTransfers,omitempty"`
+	GlobalSettingsFile          string   `json:"globalSettingsFile,omitempty"`
+	DefaultNpmRegistry          string   `json:"defaultNpmRegistry,omitempty"`
+}
+
+type BuildSettings struct {
+	MavenBuild  []BuildSettingsInfo `json:"mavenBuild,omitempty"`
+	NpmBuild    []BuildSettingsInfo `json:"npmBuild,omitempty"`
+	DockerBuild []BuildSettingsInfo `json:"dockerBuild,omitempty"`
+	MtaBuild    []BuildSettingsInfo `json:"mtaBuild,omitempty"`
+}
+
+type BuildOptions struct {
+	Profiles                    []string `json:"profiles,omitempty"`
+	GlobalSettingsFile          string   `json:"globalSettingsFile,omitempty"`
+	LogSuccessfulMavenTransfers bool     `json:"logSuccessfulMavenTransfers,omitempty"`
+	CreateBOM                   bool     `json:"createBOM,omitempty"`
+	Publish                     bool     `json:"publish,omitempty"`
+	BuildSettingsInfo           string   `json:"buildSettingsInfo,omitempty"`
+	DefaultNpmRegistry          string   `json:"defaultNpmRegistry,omitempty"`
+}
+
+func CreateBuildSettingsInfo(config *BuildOptions, buildTool string) (string, error) {
+	currentBuildSettingsInfo := BuildSettingsInfo{
+		CreateBOM:                   config.CreateBOM,
+		GlobalSettingsFile:          config.GlobalSettingsFile,
+		LogSuccessfulMavenTransfers: config.LogSuccessfulMavenTransfers,
+		Profiles:                    config.Profiles,
+		Publish:                     config.Publish,
+		DefaultNpmRegistry:          config.DefaultNpmRegistry,
+	}
+	var jsonMap map[string][]interface{}
+	var jsonResult []byte
+
+	if config.BuildSettingsInfo != "" {
+
+		err := json.Unmarshal([]byte(config.BuildSettingsInfo), &jsonMap)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to unmarshal existing build settings json '%v'", config.BuildSettingsInfo)
+		}
+
+		if build, exist := jsonMap[buildTool]; exist {
+			if reflect.TypeOf(build).Kind() == reflect.Slice {
+				jsonMap[buildTool] = append(build, currentBuildSettingsInfo)
+			}
+		} else {
+			var settings []interface{}
+			settings = append(settings, currentBuildSettingsInfo)
+			jsonMap[buildTool] = settings
+		}
+
+		jsonResult, err = json.Marshal(&jsonMap)
+		if err != nil {
+			return "", errors.Wrapf(err, "Creating build settings failed with json marshalling")
+		}
+	} else {
+		var settings []BuildSettingsInfo
+		settings = append(settings, currentBuildSettingsInfo)
+		var err error
+		switch buildTool {
+		case "mavenBuild":
+			jsonResult, err = json.Marshal(BuildSettings{
+				MavenBuild: settings,
+			})
+		case "npmBuild":
+			jsonResult, err = json.Marshal(BuildSettings{
+				NpmBuild: settings,
+			})
+		case "dockerBuild":
+			jsonResult, err = json.Marshal(BuildSettings{
+				DockerBuild: settings,
+			})
+		case "mtaBuild":
+			jsonResult, err = json.Marshal(BuildSettings{
+				MtaBuild: settings,
```

Comment:
we can also include a `case default` and return an wrapped error saying the build type is not supported , just in case we call it with an incorrect build type